### PR TITLE
PLNSRVCE-249: Update configure-build task to use sharedSecret

### PR DESCRIPTION
**Pull request by Michkov**
Use sharedSecret redhat-appstudio-user-workload as registry secret by
default. Content can be merged with the secret provided by registry-auth
workspace.

Can be merged after https://github.com/redhat-appstudio/infra-deployments/pull/349

### CODE REVIEW

File: tasks/configure-build.yaml [hunk 1]
```diff
@@ -14,28 +17,41 @@ spec:
...
 
+        DEF=/secret/default-push-secret/.dockerconfigjson
         AUTH=/workspace/registry-auth/.dockerconfigjson
         DEST=/workspace/source/.dockerconfigjson
-        echo "Looking for Registry Auth Config: $AUTH"
-        if [ -f "$AUTH" ]; then
+        echo "Looking for Registry Auth Config"
+        if [ -f "$AUTH" ] && [ -f "$DEF" ]; then
+          echo "$DEF and $AUTH found"
+          jq -M -s '.[0] * .[1]' $DEF $AUTH > $DEST
```
Comment (gabemontero) [inline tasks/configure-build.yaml:43]: slick ;-) .. if my refresher on `jq` help/man pages and stackoverflow (https://stackoverflow.com/questions/42011086/merge-arrays-of-json) are correct, this is combining the .dockerconfigjson's, where the `-s` is key, right?

**Review by gabemontero — COMMENTED** [review]
A couple of minor questions that most likely fall under the "educate Gabe" category @Michkov 

but I'll wait for the answer to the "should we exit immediately when no auth is present" question before officially approving

otherwise LGTM

thanks

File: tasks/configure-build.yaml [hunk 2]
```diff
@@ -14,28 +17,41 @@ spec:
...
+        elif [ -f "$DEF" ]; then
+          echo "$DEF found"
+          cp $DEF $DEST
         else
-          echo "No $AUTH found."
+          echo "No auth found."
           echo -n " " > /tekton/results/registry-auth
           echo -n " " > /tekton/results/buildah-auth-param
-          echo
+          exit 0
```
[the remark this answers is not in the document]
Comment (gabemontero) [reply tasks/configure-build.yaml:54]: ah yeah I remember the mention of imagestream usage in other discussions / demos

good to know that connects to this part of the system

thanks

[the remark this answers is not in the document]
Comment (gabemontero) [reply tasks/configure-build.yaml:54]: maybe a comment here mentioning that case and why we exit cleanly here

**Comment by Michkov** [comment]
@gabemontero Improved secret handling, now taking also secrets attached to ServiceAccount

**Pull request merged**